dashboard/entry/tagcheck.go

```diff
@@ -0,0 +1,43 @@
+package entry
+
+import (
+	"fmt"
+
+	"github.com/jinzhu/gorm"
+	"github.com/traggo/server/generated/gqlmodel"
+	"github.com/traggo/server/model"
+)
+
+func tagsDuplicates(src []*gqlmodel.InputTimeSpanTag, dst []*gqlmodel.InputTimeSpanTag) *gqlmodel.InputTimeSpanTag {
+	existingTags := make(map[string]struct{})
+	for _, tag := range src {
+		existingTags[tag.Key+":"+tag.Value] = struct{}{}
+	}
+
+	for _, tag := range dst {
+		if _, ok := existingTags[tag.Key+":"+tag.Value]; ok {
+			return tag
+		}
+
+		existingTags[tag.Key] = struct{}{}
+	}
+
+	return nil
+}
+
+func tagsExist(db *gorm.DB, userID int, tags []*gqlmodel.InputTimeSpanTag) error {
+	existingTags := make(map[string]struct{})
+
+	for _, tag := range tags {
+		if _, ok := existingTags[tag.Key]; ok {
+			return fmt.Errorf("tag '%s' is present multiple times", tag.Key)
```

Comment:
Why is this an error? It seems reasonable to include multiple tag:value pairs where tag is used multiple times. E.g. includeTag: ['project:traggo', 'project:gotify']